src/message.rs

```diff
@@ -17,9 +18,127 @@ use crate::pgp::*;
 use crate::sql;
 use crate::stock::StockMessage;
 
-/// In practice, the user additionally cuts the string himself pixel-accurate.
+// In practice, the user additionally cuts the string themselves
+// pixel-accurate.
 const SUMMARY_CHARACTERS: usize = 160;
 
+/// Message ID, including reserved IDs.
+///
+/// Some message IDs are reserved to identify special message types.
+/// This type can represent both the special as well as normal
+/// messages.
+#[derive(Debug, Copy, Clone, Default, PartialEq, Eq)]
+pub struct MsgId(u32);
+
+impl MsgId {
+    /// Create a new [MsgId].
+    pub fn new(id: u32) -> MsgId {
+        MsgId(id)
+    }
+
+    /// Whether the message ID signifies a special message.
+    ///
+    /// This kind of message ID can not be used for real messages.
+    pub fn is_special(&self) -> bool {
+        match self.0 {
+            0..=DC_MSG_ID_LAST_SPECIAL => true,
+            _ => false,
```

Comment:
instead of this slightly odd range match expression `self.0 < DC_MSG_ID_LAST_SPECIAL + 1` should suffice.  Is the range pattern match here a stable feature?

Reply:
Yeah, I also realised that these weird contortions to make match work are not worth it and if..else if is much nicer.

Reply:
So apparently I left this one in here.  The compiler told me that a bare `..` for the range was deprecated and to use `..=` instead.  I think this is because they want to add `...` as a range specifier which does not include the first and last number given, trying to use `...` now gives you a warning that it is not stable, while for `..=` this warning does not exist so I think this is fine.